### DISCUSSION TITLE
Lite

### DISCUSSION THREAD

**Discussion by adjourn** [pmndrs/react-spring#1323] [Support]
Do you see this library ever getting a lite bundle with few sacrifices here and there?

For example I got almost 20% size reduction by simply commenting out [named colors map](https://github.com/react-spring/react-spring/blob/master/src/shared/colors.ts) and replacing it with empty map `8.44Kb -> 6.87Kb (gzip)`. It means I can't use named colors like `lemonchiffon` but seriously, does anyone even use them?

**Comment by drcmda** [comment]
you could brew your own, or use /universal, which is pretty slim (and it doesn't include colors).

    import { useSpring, animated, ... } from 'react-spring/universal'
    import { Spring, animated, ... } from 'react-spring/renderprops-universal'

the full export from this will be 6.9KB, but that's without tree shaking. if you use only useSpring for instance, you'll end up with 4.7kb.

**Comment by jacobrask** [comment]
I did some looking in the styled-components repo because I figured there is some overlap in some constant arrays, and see that you've already suggested some sharing in https://github.com/styled-components/styled-components/issues/1688 but it seems like they plan to solve it with a babel transform instead. 

Splitting those constants out to separate packages would make it simpler for really size-concious users to alias them to empty arrays in their build systems though, in case they still want to use the other features from the web target.

**Comment by drcmda** [comment]
having a babel plug would be fantastic, not only for colors but also animated.div, etc. there's some weight it has to drag along for the dom stuff.

**Comment by adjourn** [comment]
I just recently made a inhouse Babel plugin so the API is still fresh in my memory.

@drcmda If you could give me few pointers or compile a list of links to variables/code that is safe to remove, I could give it a shot sometimes soon.

**Comment by drcmda** [comment]
Basically all colors and native elements (animated.xxxx) could be white listed by Babel. Though colors for sure would be harder to implement.
